lib/elixir/unicode/tokenizer.ex

```diff
@@ -138,10 +162,11 @@ defmodule String.Tokenizer do
       with [range, scripts_with_comments] <- :binary.split(line, ";"),
            [scripts, _comments] <- :binary.split(scripts_with_comments, "#"),
            scripts =
-             scripts |> String.split(" ", trim: true) |> Enum.map(&Map.get(aliases, &1, &1)),
-           true <- "Common" not in scripts and "Inherited" not in scripts do
+             scripts |> String.split(" ", trim: true) |> Enum.map(&Map.get(aliases, &1, &1)) do
         for codepoint <- range_to_codepoints.(range),
-            Map.has_key?(unicode_all, codepoint),
+            (Map.has_key?(unicode_all, codepoint) and
+               "Common" not in scripts and "Inherited" not in scripts) or
+              Map.has_key?(normalizations, codepoint),
```

Comment:
I think we can remove this line? Don't we default to `top` for anything that is not included?

Reply:
We default to `bottom` for anything filtered out here -- in the sense that, if filtered out here, it won't be used in the clauses that generate `unicode_start`, `unicode_continue`. So any filtered-out codepoints, when encountered in code, will fall through those to the catch-all clause that returns `bottom`, the empty scriptset, which catches all unsupported codepoints (regardless of whether it's because they're the wrong category, like not ID Start/Continue, or because they're Restricted, etc etc).

I *might* know a clause you're thinking about -- near the end of `tokenizer.ex`, there's a `with` expression that gets a character's scriptset, and it's written in a way that implies that if neither `unicode_{upper,start,continue}` return a scriptset, that it should default to `top` -- however I think in practice it won't since it can never be given an input like that, as it's called from validation, after the token is already parsed.